go/cmake/golang.cmake

```diff
@@ -10,7 +10,8 @@ function(GO_LIBRARY NAME BUILD_TYPE)
   else()
     set(BUILD_MODE -buildmode=c-shared)
     if(APPLE)
-      set(LIB_NAME "lib${NAME}.dylib")
+#      set(LIB_NAME "lib${NAME}.dylib")
+      set(LIB_NAME "lib${NAME}.so")
```

Comment:
compile library with `.so` suffix in Apple system, it will still generate apple arch binary but with `.so` suffix, is it runnable?